cloudify_cli/bootstrap/bootstrap.py

```diff
@@ -118,22 +120,64 @@ def bootstrap(blueprint_path,
                 task_retry_interval=task_retry_interval,
                 task_thread_pool_size=task_thread_pool_size)
 
+    nodes = env.storage.get_nodes()
     node_instances = env.storage.get_node_instances()
-    nodes_by_id = {node.id: node for node in env.storage.get_nodes()}
-    manager_node_instance = \
-        next(node_instance for node_instance in node_instances if
-             'cloudify.nodes.CloudifyManager' in
-             nodes_by_id[node_instance.node_id].type_hierarchy)
-    provider_context = \
-        manager_node_instance.runtime_properties[PROVIDER_RUNTIME_PROPERTY]
-    manager_ip = \
-        manager_node_instance.runtime_properties[MANAGER_IP_RUNTIME_PROPERTY]
-    manager_user = \
-        manager_node_instance.runtime_properties[MANAGER_USER_RUNTIME_PROPERTY]
-    manager_key_path = manager_node_instance.runtime_properties[
-        MANAGER_KEY_PATH_RUNTIME_PROPERTY]
-    rest_port = \
-        manager_node_instance.runtime_properties[REST_PORT]
+    nodes_by_id = {node.id: node for node in nodes}
+
+    try:
+        manager_node_instance = \
+            next(node_instance for node_instance in node_instances if
+                 'cloudify.nodes.CloudifyManager' in
+                 nodes_by_id[node_instance.node_id].type_hierarchy)
+    except Exception:
+        manager_node_instance = \
+            next(node_instance for node_instance in node_instances if
+                 'cloudify.nodes.MyCloudifyManager' in
+                 nodes_by_id[node_instance.node_id].type_hierarchy)
+        manager_node = nodes_by_id['manager_configuration']
+
+    if manager_node_instance.runtime_properties.get('provider'):
+        provider_context = \
+            manager_node_instance.runtime_properties[
+                PROVIDER_RUNTIME_PROPERTY]
+        manager_ip = \
+            manager_node_instance.runtime_properties[
+                MANAGER_IP_RUNTIME_PROPERTY]
+        manager_user = \
+            manager_node_instance.runtime_properties[
+                MANAGER_USER_RUNTIME_PROPERTY]
+        manager_key_path = manager_node_instance.runtime_properties[
+            MANAGER_KEY_PATH_RUNTIME_PROPERTY]
+        rest_port = \
+            manager_node_instance.runtime_properties[REST_PORT]
+    else:
+        manager_ip = \
+            manager_node_instance.runtime_properties['manager_host_public_ip']
+        manager_user = manager_node.properties['ssh_user']
+        manager_key_path = manager_node.properties['ssh_key_filename']
+        rest_port = 80
+
+        # these should be changed to allow receiving the
+        # paths from the blueprint.
+        agent_remote_key_path = manager_node.properties.get(
+            'agent_remote_key_path', constants.AGENT_REMOTE_KEY_PATH)
+        agent_local_key_path = manager_node.properties.get(
+            'agent_local_key_path', constants.AGENT_LOCAL_KEY_PATH)
+        fabric_env = {
```

Comment:
We currently return the key_path to the cli for cfy ssh and the likes. Adding a password now might incur unknown behavior.